src/wrappers/cobs.py

```diff
@@ -1,15 +1,97 @@
 from os import environ
+from pathlib import Path
+from tempfile import TemporaryDirectory
 
 import cobs_index as cobs
 
-
-COBS_INDEX_PATH = environ.get('COBS_INDEX_PATH', '/data/500.cobs_compact')
+COBS_CLASSIC_INDEXES_DIR = environ.get('COBS_CLASSIC_INDEXES_DIR', '/data/classic')
+COBS_SAMPLE_DIR = environ.get('COBS_SAMPLE_DIR', '/data/samples')
+COBS_CLASSIC_FILE_EXTENSION = 'cobs_classic'
 COBS_TERM_SIZE = 31
+COBS_POSITIVE_RATE = 0.4
 
 
 class Cobs:
-    def __init__(self, index_path=COBS_INDEX_PATH):
-        self.search_instance = cobs.Search(index_path)
+    def __init__(self):
+        index_paths = []
+        for path in Path(COBS_CLASSIC_INDEXES_DIR).iterdir():
+            if path.is_dir():
+                index_paths += [p for p in path.iterdir() if p.is_file()]
+        self.search_instances = [cobs.Search(str(index_path)) for index_path in index_paths]
 
     def search(self, query, threshold):
-        return self.search_instance.search(query, threshold=threshold)
+        results = []
+
+        for search_instance in self.search_instances:
+            results += search_instance.search(query, threshold=threshold)
+
+        flatten = []
+        for r in results:
+            flatten.append((r.score, r.doc_name))
+        return flatten
+
+    @staticmethod
+    def build(sample_paths, sample_names):
+        assert len(sample_paths) == len(sample_names)
+
+        p = cobs.ClassicIndexParameters()
+        p.term_size = COBS_TERM_SIZE
+        p.false_positive_rate = COBS_POSITIVE_RATE
+        setattr(p, 'continue', True)
+
+        Cobs.group_samples_by_signature_size(sample_paths, sample_names, p)
+        Cobs.build_new_indexes(p)
+        Cobs.combine_classic_indexes(p)
+
+        for sample_dir in Path(COBS_SAMPLE_DIR).iterdir():
+            for sample in sample_dir.glob('*'):
+                sample.unlink()
+
+    @staticmethod
+    def group_samples_by_signature_size(sample_paths, sample_names, params):
+        # cobs.DocumentEntry doesn't have a constructor, so the only way to construct one is to iterate a DocumentList
+        with TemporaryDirectory() as tmpdir:
+            doclist = cobs.DocumentList(tmpdir)
+            for original_sample_path in sample_paths:
+                doclist.add(original_sample_path.strip())
+
+            # Copy sample file to the group with the smallest signature size that is larger than its own
+            for doc, name in zip(doclist, sample_names):
+                signature_size = cobs.calc_signature_size(doc.num_terms(COBS_TERM_SIZE), params.num_hashes, params.false_positive_rate)
+                for index_dir in Path(COBS_SAMPLE_DIR).iterdir():
+                    if index_dir.is_dir() and int(index_dir.name) > signature_size:
+                        file_extension = doc.path.split('.')[-1]
+                        new_sample_path = index_dir / (name + '.' + file_extension)
+                        shutil.copyfile(doc.path, new_sample_path)
```

Comment:
Can we not copy files? I don't think it is absolutely necessary.

Reply:
I was thinking we could copy in case sth go wrong the original samples are untouched. Then we can have a cronjob to backup the samples elsewhere and clean up the directories periodically

Reply:
We don't expect to modify the original input. What we do is purely readonly access. Leave the cleanup responsibility to the client.